editoast/editoast_derive/src/modelv2.rs

```diff
@@ -0,0 +1,648 @@
+                            .set(&self)
+                            .get_result::<#row_ident>(conn)
+                            .await
+                            .map(Into::into)
+                            .optional()
+                            .map_err(Into::into)
+                    }
+                }
+
+                #[automatically_derived]
+                #[async_trait::async_trait]
+                impl crate::modelsv2::DeleteStatic<#ty> for #model {
+                    async fn delete_static(
+                        conn: &mut diesel_async::AsyncPgConnection,
+                        #ident: #ty,
+                    ) -> crate::error::Result<bool> {
+                        use diesel::prelude::*;
+                        use diesel_async::RunQueryDsl;
+                        use #table_mod::dsl;
+                        diesel::delete(dsl::#table_name.#filter)
+                            .execute(conn)
+                            .await
+                            .map(|n| n == 1)
+                            .map_err(Into::into)
+                    }
+                }
+            )*
+
+            #[automatically_derived]
+            #[async_trait::async_trait]
+            impl crate::modelsv2::Create<#model> for #cs_ident {
+                async fn create(
+                    self,
+                    conn: &mut diesel_async::AsyncPgConnection,
+                ) -> crate::error::Result<#model> {
+                    use diesel_async::RunQueryDsl;
+                    diesel::insert_into(#table_mod::table)
+                        .values(&self)
+                        .get_result::<#row_ident>(conn)
+                        .await
+                        .map(Into::into)
+                        .map_err(Into::into)
+                }
+            }
+
+            #[automatically_derived]
+            #[async_trait::async_trait]
+            impl crate::modelsv2::Delete for #model {
+                async fn delete(
+                    &self,
+                    conn: &mut diesel_async::AsyncPgConnection,
+                ) -> crate::error::Result<bool> {
+                    use diesel::prelude::*;
+                    use diesel_async::RunQueryDsl;
+                    use #table_mod::dsl;
+                    let id = self.#pk;
+                    diesel::delete(#table_mod::table.find(id))
+                        .execute(conn)
+                        .await
+                        .map(|n| n == 1)
+                        .map_err(Into::into)
+                }
+            }
+        }
+    }
+}
+
+impl GeneratedType {
+    fn ident(&self) -> Ident {
+        Ident::new(self.type_name.as_ref().unwrap(), Span::call_site())
+    }
+}
+
+pub fn model(input: &DeriveInput) -> Result<TokenStream> {
+    let model_name = &input.ident;
+    let model_vis = &input.vis;
+    let options = ModelOptions::from_derive_input(input).expect("Model: bad options");
```

Comment:
```suggestion
    let options = ModelOptions::from_derive_input(input)?;
```

Avoid panicking the macro.